syft/grid/grid_client.py

```diff
@@ -75,14 +82,96 @@ def _send_http_req(self, method, path: str, params: dict = None, body: bytes = N
         if method == "GET":
             res = requests.get(self.http_url + path, params)
         elif method == "POST":
-            res = requests.post(self.http_url + path, body)
+            res = requests.post(self.http_url + path, params=params, data=body)
 
         if not res.ok:
             raise GridError("HTTP response is not OK", res.status_code)
 
         response = res.content
         return response
 
+    def _yield_chunk_from_request(self, request, chunk_size):
+        for chunk in request.iter_content(chunk_size=chunk_size):
+            yield chunk
+
+    def _read_n_request_chunks(self, chunk_generator, n):
+        for i in range(n):
+            try:
+                next(chunk_generator)
+            except:
+                return False
+        return True
+
+    def _get_ping(self, worker_id, random_id):
+        params = {"is_ping": 1, "worker_id": worker_id, "random": random_id}
+        ping = (
+            timeit(
+                lambda: self._send_http_req("GET", "/federated/speed-test", params),
+                number=MAX_SPEED_TESTS,
+            )
+            * 1000
+        )  # for ms
+        return ping
+
+    def _get_upload_speed(self, worker_id, random_id):
+        buffer_size = CHUNK_SIZE
+        speed_history = []
+
+        for _ in range(MAX_SPEED_TESTS):
+            data_sample = b"x" * buffer_size
+            params = {"worker_id": worker_id, "random": random_id}
+            body = {"upload_data": data_sample}
+            time_taken = timeit(
+                lambda: self._send_http_req("POST", "/federated/speed-test", params, body),
+                number=1,
+            )
+            if time_taken < 0.5:
+                buffer_size = min(
+                    buffer_size * SPEED_MULT_FACTOR, MAX_BUFFER_SIZE * 64
```

Comment:
MAYBE: Use constant variable for 64 as there is for the others? (also it seems like the buffer size is actually 64 * 1048576). Maybe it would be better to use another name for ```MAX_BUFFER_SIZE```?